File: programmers/javascript/level2/프린터.js
const priorities = [1, 1, 9, 1, 1, 1];
const location = 0;

function solution(priorities, location) {
  var answer = 0;
  var arr = [];
  for (let i = 0; i < priorities.length; i++) {
    arr.push([priorities[i], i]);
  }
  while (true) {
    let max_num = 0;
    for (let i = 0; i < arr.length; i++) {
      if (max_num < arr[i][0]) {
        max_num = arr[i][0];
      }
    }
    if (max_num === arr[0][0]) {
      answer++;
      if (location === arr[0][1]) return answer;
      arr.splice(0, 1);
    } else {
      let temp = arr[0];
      arr.splice(0, 1);
      arr.push(temp);
    }
  }
}

console.log(solution(priorities, location));

File: python/programmers/level2/프린터.py
from collections import deque


def solution(priorities, location):
    answer = 1
    indexArr = deque([i for i in range(len(priorities))])
    q = deque(priorities)
    while True:
        maxnum = max(q)
        # print(maxnum)
        while maxnum != q[0]:
            tmp = q.popleft()
            q.append(tmp)
            tmp = indexArr.popleft()
            indexArr.append(tmp)
        if indexArr[0] != location:
            answer += 1
            q.popleft()
            indexArr.popleft()
            # print(q)
        else:
            break

    return answer


print(solution([2, 1, 3, 2], 2))


# def solution(priorities, location):
#     queue =  [(i,p) for i,p in enumerate(priorities)]
#     answer = 0
#     while True:
#         cur = queue.pop(0)
#         if any(cur[1] < q[1] for q in queue):
#             queue.append(cur)
#         else:
#             answer += 1
#             if cur[0] == location:
#                 return answer